pkg/filters/proxies/grpcproxy/proxy.go

```diff
@@ -76,12 +88,22 @@ type (
 
 		mainPool       *ServerPool
 		candidatePools []*ServerPool
+		pool           *MultiPool
+		poolSpec       *objectpool.Spec
```

Comment:
please find a better name, it is too confusion as `mainPool` & `candidatePools` are already there.